packages/puppeteer-core/src/api/ElementHandle.ts

```diff
@@ -886,7 +886,24 @@ export class ElementHandle<
    * or `null` if the element is not visible.
    */
   async boundingBox(): Promise<BoundingBox | null> {
-    throw new Error('Not implemented');
+    if (this.frame.parentFrame()) {
+      throw new Error(
+        'Elements within nested iframes are currently not supported.'
+      );
+    }
+    const box = await this.evaluate(element => {
```

Comment:
is it in the main word or in the puppeteer world?

Reply:
Main world.

Reply:
But that shouldn't matter.

Reply:
Could you add a todo that it should be done in the isolated world?

Reply:
It matters for the edge cases because the page can override getBoundingClientRect

Reply:
Hmm, gotcha. Let me switch to the puppeteer world.

Reply:
Forgot that we can't because we don't have sandboxes yet ;)

I've added a TODO.

Reply:
we do have sandboxes:

```
this.frame.isolatedRealm()
```